### DISCUSSION TITLE
Are Power Profile GUIDs Consistent?

### DISCUSSION THREAD

**Discussion by hockinsk** [seerge/g-helper#2834] [General]
I'm troubleshooting why when in Turbo mode from G-Helper performance is seriously reduced. Should my Power Config GUIDs be consistent with G-Helper. When I list my options I have:

```
>powercfg /list

Existing Power Schemes (* Active)
-----------------------------------
Power Scheme GUID: 1a5fb54c-bb93-4eac-9b92-7088f607b028  (Power Saver)
Power Scheme GUID: 381b4222-f694-41f0-9685-ff5bb260df2e  (Balanced) *
Power Scheme GUID: c70ae11f-fedb-4a4b-b924-5e1d86ea188a  (Ultimate Performance)
Power Scheme GUID: ec34dd3e-1364-406d-a6e1-fac63fadf157  (High Performance)
```

On the [readme page, G-Helper lists the following is set though?](https://github.com/seerge/g-helper/wiki/Power-user-settings) page it lists the following GUIDs are used.

```
"scheme_0": "381b4222-f694-41f0-9685-ff5bb260df2e", (Balanced)
"scheme_1": "8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c", (High)
"scheme_2": "381b4222-f694-41f0-9685-ff5bb260df2e", (Ultimate)
```

I notice only Balanced is the same GUID and when switching between Silent, Balanced and Turbo, G-Helper only seems to switch to Balanced. When I use Ultimate, something definitely changes in CPU performance even though Windows shows the Power Plan as Balances still, but it changes for the worse. e.g. here's a DSP graph of my DAW and when switching to Turbo mode in G-Helper, the DSP maxes out the CPU and sound is inaudible.

![image](https://github.com/user-attachments/assets/3e6e137c-a4cd-42a4-9cf9-a6ccce882a54)

Should I be setting the GUIDS in g-helper config file to match the GUIDS of Windows Power Plans?

Thanks.

**Comment by seerge** [comment]
@hockinsk hello, I used GUIDs from [Microsoft docs](https://learn.microsoft.com/en-us/windows/win32/power/power-policy-settings). But they also match power plan GUIDS on my device (G14 2022, Windows 11).

If power plans have different GUIDs on your machine by some reason, you should probably use them. I just wonder why, cause googling ``ec34dd3e-1364-406d-a6e1-fac63fadf157`` doesn't bring up anything :)

Or did you just copy default power plans into something custom on your system ?

--

And PS ``"scheme_2": "381b4222-f694-41f0-9685-ff5bb260df2e", (Ultimate)`` -> is not Ultimate, as that's GUID of default Balanced plan :)

**Reply by hockinsk** [reply]
It's a laptop, I used a csenabled tool, so maybe it creates its own guids rather than Microsofts?
I seem to have resolved it now by setting the GUIDS in the json config to those in powercfg. In Fans & Power of G-Helper I set CPU Boost to disabled and that has stabilised my DAW audio and Turbo giving some extra stability to the DSP too. Typically I would set the Min/Max Processor state for the CPU to 99% which usually ahs the effect of disabling the CPU going into Turbo mode and being unstable for Audio. All good now I think, thanks. I love this G-Helper!

**Reply by seerge** [reply]
@hockinsk of course it's a laptop. 

I don't know what `csenabled tool ` is, but if power plans on your device have different GUIDs and you want to use that specific power plans - you need to use them.

I'm glad you have resolved your issue.

**Reply by hockinsk** [reply]
This was the tool. My Asus ROG Flow Z13 didn't come with any Power Plans, so I used that tool to add them. Seems to work well.
https://www.youtube.com/watch?v=E_-Tv_e0zLY

**Reply by seerge** [reply]
@hockinsk i see, i think what it does is simply copies those power plans so they become visible in Control Panel.  
Most probably original plans with original GUIDs also exist in your system as they should :)
